pkg/k8s/factory_functions.go

```diff
@@ -204,10 +204,24 @@ func EqualV1Pod(pod1, pod2 *v1.Pod) bool {
 }
 
 func EqualV1Node(node1, node2 *v1.Node) bool {
-	// The only information we care about the node is it's annotations, in
-	// particularly the CiliumHostIP annotation.
-	return node1.GetObjectMeta().GetName() == node2.GetObjectMeta().GetName() &&
-		node1.GetAnnotations()[annotation.CiliumHostIP] == node2.GetAnnotations()[annotation.CiliumHostIP]
+	if node1.GetObjectMeta().GetName() != node2.GetObjectMeta().GetName() {
+		return false
+	}
+
+	anno1 := node1.GetAnnotations()
+	anno2 := node2.GetAnnotations()
+	annotationsWeCareAbout := []string{
```

Comment:
Can you declare this slice outside of this function? Otherwise we are allocating it for every `EqualV1Node` invocation.

Reply:
Hrm, is this really how it works? The compiler should know the size at compile time and it's not passed around so I'd expect it to just be put together on the stack.

Reply:
🤔 now I'm not sure